src/azure-cli/azure/cli/command_modules/vm/_actions.py

```diff
@@ -40,15 +40,28 @@ def load_images_thru_services(cli_ctx, publisher, offer, sku, location):
         location = get_one_of_subscription_locations(cli_ctx)
 
     def _load_images_from_publisher(publisher):
-        offers = client.virtual_machine_images.list_offers(location, publisher)
+        from msrestazure.azure_exceptions import CloudError
+        try:
+            offers = client.virtual_machine_images.list_offers(location, publisher)
+        except CloudError as e:
+            logger.warning(str(e))
```

Comment:
Previously, if there is an exception, it will be treated as an error. After updating, it will be treated as a warning, right ?

Reply:
Yes. There are many publishers. This function will be invoked many times. An error will fail the command. But there are other valid publishers. So I catch it.